config/module.go

```diff
@@ -115,46 +115,97 @@ func (m Module) NeedsSyntheticPackage() bool {
 	return m.Type == ModuleTypeLocal && tarballExtensionsRegexp.MatchString(strings.ToLower(m.ExePath))
 }
 
-// SyntheticPackage creates a fake package for a local module which points to a local tarball.
+// SyntheticPackage creates a fake package for a module which can be used to access some package logic.
 func (m Module) SyntheticPackage() (PackageConfig, error) {
-	var ret PackageConfig
 	if m.Type != ModuleTypeLocal {
-		return ret, errors.New("non-local package passed to syntheticPackage")
+		return PackageConfig{}, errors.New("SyntheticPackage only works on local modules")
 	}
-	ret.Name = fmt.Sprintf("synthetic-%s", m.Name)
-	ret.Package = ret.Name
-	ret.Type = PackageTypeModule
-	return ret, nil
+	var name string
+	if m.NeedsSyntheticPackage() {
+		name = fmt.Sprintf("synthetic-%s", m.Name)
+	} else {
+		name = m.Name
+	}
+	return PackageConfig{Name: name, Package: name, Type: PackageTypeModule}, nil
 }
 
-// syntheticPackageExeDir returns the unpacked ExePath for local tarball modules.
-func (m Module) syntheticPackageExeDir(packagesDir string) (string, error) {
+// exeDir returns the parent directory for the unpacked module.
+func (m Module) exeDir(packagesDir string) (string, error) {
+	if !m.NeedsSyntheticPackage() {
+		return filepath.Dir(m.ExePath), nil
+	}
 	pkg, err := m.SyntheticPackage()
 	if err != nil {
 		return "", err
 	}
 	return pkg.LocalDataDirectory(packagesDir), nil
 }
 
-// EvaluateExePath returns absolute ExePath except for local tarballs where it looks for side-by-side meta.json.
-// The side-by-side lookup is because we don't bundle entrypoint in module tarballs, it's not an intentional design choice.
+// parseJSONPath returns a *T by parsing the json file at `path`.
+func parseJSONPath[T any](path string) (*T, error) {
```

Comment:
Might just be me not having enough coffee this morning, but I confused with https://github.com/json-path/JsonPath
```suggestion
// parseJSONFilePath returns a *T by parsing the json file at `path`.
func parseJSONFilePath[T any](path string) (*T, error) {
```